HomeWork2/Clean Architecture/Catalog Service/Infrastructure/Interfaces/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Application.Interfaces.Repositories;
+using Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Interfaces.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : BaseEntity
```

Comment:
You can't name this class "Generic" because it is not. Ot it should be at least EFGenericRepository

Reply:
changed